pkg/metrics/interfaces.go

```diff
@@ -37,7 +37,7 @@ var (
 
 	NoOpCounter     prometheus.Counter     = &counter{NoOpMetric, NoOpCollector}
 	NoOpCounterVec  CounterVec             = &counterVec{NoOpCollector}
-	NoOpObserver    prometheus.Observer    = &observer{}
+	NoOpObserver    prometheus.Histogram   = &observer{NoOpCollector}
```

Comment:
We need a type that satisfies both `prometheus.Collector` and `prometheus.Observer`. The first because [we group](https://github.com/cilium/cilium/blob/master/pkg/metrics/metrics.go#L687) all metrics together in a slice of `prometheus.Collector`. The second because the `prometheus.Observer` interface contains the `Observe` method we use to report the metric in `cilium_cidr_group.go`.

Probably it is better to add a new type called NoOpHistogram to avoid confusion with the name. I'll fix it.